lib/request.js

```diff
@@ -161,7 +162,10 @@ Object.defineProperties(Request.prototype, {
   },
   headers: {
     get () {
-      return this.raw.headers
+      return this.processedHeaders || this.raw.headers
```

Comment:
For this to be accurate, `this.processedHeaders` would need to be initialized with all of the headers from `this.raw.headers`.